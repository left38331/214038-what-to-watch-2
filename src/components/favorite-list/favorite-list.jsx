import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';

import {Logo} from 'components/logo/logo';
import {UserBlock} from 'components/user-block/user-block';

const FavoriteList = (props) => {
  return <div className="user-page">
    <header className="page-header user-page__head">
      <Logo/>

      <h1 className="page-title user-page__title">My list</h1>

      <UserBlock
        isAuthorizationRequired={props.isAuthorizationRequired}
        avatar={props.avatar}
      />
    </header>

    <section className="catalog">
      <h2 className="catalog__title visually-hidden">Catalog</h2>

      <div className="catalog__movies-list">
        {props.favoriteFilms.map((film, i) => {
          return <article className="small-movie-card catalog__movies-card" key={i}>
            <div className="small-movie-card__image">
              <img src={film.backgroundImage} alt={film.title} width="280" height="175"/>
            </div>
            <h3 className="small-movie-card__title">
              <Link to={`/films/${film.id}`} className="small-movie-card__link">{film.title}</Link>
            </h3>
          </article>;
        })}

      </div>
    </section>

    <footer className="page-footer">
      <Logo
        isFooter={true}
      />

      <div className="copyright">
        <p>Â© 2019 What to watch Ltd.</p>
      </div>
    </footer>
  </div>;
};

FavoriteList.propTypes = {
  isAuthorizationRequired: PropTypes.bool.isRequired,
  avatar: PropTypes.string.isRequired,
  favoriteFilms: PropTypes.array.isRequired,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  isAuthorizationRequired: state.isAuthorizationRequired,
  avatar: state.avatarUrl,
  favoriteFilms: state.favoriteFilms
});

export {FavoriteList};
export default connect(mapStateToProps)(FavoriteList);
