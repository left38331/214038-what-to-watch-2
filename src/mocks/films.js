export const films = [
  {
    id: 1,
    title: `Fantastic Beats`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    alt: `Fantastic Beasts: The Crimes of Grindelwald`,
    link: `movie-page.html`,
    preview: `https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/trailers/fbwft-rutlr3_854.mp4`,
    genre: `Fantasy`
  },
  {
    id: 2,
    title: `Bohemian Rhapsody`,
    poster: `img/bohemian-rhapsody.jpg`,
    alt: `Bohemian Rhapsody`,
    link: `movie-page.html`,
    preview: `https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/media/video/c/76/c766c75666111cab4aa6bb75a5424bc0.480.mp4`,
    genre: `Drama`
  },
  {
    id: 3,
    title: `Macbeth.jpg`,
    poster: `img/macbeth.jpg`,
    alt: `Macbeth`,
    link: `movie-page.html`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Drama`
  },
  {
    id: 4,
    title: `Aviator`,
    poster: `img/aviator.jpg`,
    alt: `Aviator`,
    link: `movie-page.html`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `Adventure`
  },
  {
    id: 5,
    title: `War of the worlds`,
    poster: `img/war-of-the-worlds.jpg`,
    alt: `War of the worlds`,
    link: `movie-page.html`,
    preview: `https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/media/video/5/2a/52ab2988c90e81715008ed5189632b8d.480.mp4`,
    genre: `Action`
  },
];
