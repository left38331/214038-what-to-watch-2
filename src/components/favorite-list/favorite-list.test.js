import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import {FavoriteList} from 'components/favorite-list/favorite-list';
import {films} from '../../mocks/films';

const props = {
  isAuthorizationRequired: false,
  avatar: `/img/egor-forever-yang.jpg`,
  favoriteFilms: films
};

it(`Render correctly favorite-film block component`, () => {
  const favoriteBlock = renderer.create(<BrowserRouter><FavoriteList {...props} /></BrowserRouter>).toJSON();

  expect(favoriteBlock).toMatchSnapshot();
});
