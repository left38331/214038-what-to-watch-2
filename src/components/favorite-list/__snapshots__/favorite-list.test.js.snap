// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly favorite-film block component 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link "
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        className="user-block__avatar"
        href="/mylist"
        onClick={[Function]}
      >
        <img
          alt="User avatar"
          height="63"
          src="https://htmlacademy-react-2.appspot.com//img/egor-forever-yang.jpg"
          width="63"
        />
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="Fantastic Beats"
            height="175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/1"
            onClick={[Function]}
          >
            Fantastic Beats
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="Bohemian Rhapsody"
            height="175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/2"
            onClick={[Function]}
          >
            Bohemian Rhapsody
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="Macbeth.jpg"
            height="175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/3"
            onClick={[Function]}
          >
            Macbeth.jpg
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="Aviator"
            height="175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/4"
            onClick={[Function]}
          >
            Aviator
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="War of the worlds"
            height="175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/5"
            onClick={[Function]}
          >
            War of the worlds
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
